cmd/login/oidc.go

```diff
@@ -111,6 +113,36 @@ func handleOIDC(ctx context.Context, out io.Writer, errOut io.Writer, i *install
 	return authResult, nil
 }
 
+// handleDeviceFlowOIDC executes the OIDC device authentication flow against an installation's authentication provider.
+func handleDeviceFlowOIDC(out io.Writer, in *os.File, i *installation.Installation) (authInfo, error) {
+	auther := oidc.NewDeviceAuthenticator(clientID, i)
+
+	deviceCodeData, err := auther.LoadDeviceCode()
+	if err != nil {
+		return authInfo{}, microerror.Mask(err)
+	}
+
+	_, _ = fmt.Fprintf(out, "Open this URL in the browser to log in:\n%s\nPress ENTER when ready to continue\n", deviceCodeData.VerificationUriComplete)
```

Comment:
Interesting. I also tested it on Golem and it worked for me. Maybe it behaves differently on different operating systems? I will investigate.

Regarding the context name - it is intentional. Kubectl-gs relies heavily on context name and derives it from various input data, so I thought it would be safer to change the user name and keep the same context name. It may be possible to change the context name as well, but then it might be necessary to make adjustments in many other places of kgs.

When Dex generates the device code, it also provides an information about how long it's valid and how frequently it can be polled. So, it should be possible to use it to avoid the wait for user input. We could poll Dex for the state of the authentication periodically and fail in case the device code expiration period is reached/exceeded.

In case the above 2 changes are added to this PR, it may grow in size quite a bit. Is that OK?

Reply:
If you think it'll be a lot, how about a separate PR?
The implementation as it is right now already adds value so this isn't a blocker to me. :) I was thinking more about the future utility.

Reply:
It's a good point. I haven't thought about such a scenario (using kubectl-gs in a completely non-interactive environment). Of course, the obvious use case would be the usage in automation pipelines. But since Dex still requires to open a browser and go through the OAuth flow while using the device flow, I did not consider it to be usable in such cases. But maybe there is a way.

I can give it a try and we'll see how big the change will be.

Reply:
The changes are done. Now the device auth flow does not require user interaction (other than completing the in-browser authentication).